flocker/control/httpapi.py

```diff
@@ -1025,7 +1046,38 @@ def api_dataset_from_dataset_and_node(dataset, node_uuid):
     return result
 
 
-def create_api_service(persistence_service, cluster_state_service, endpoint):
+def tls_user_context_factory(path, username=u"client"):
```

Comment:
This is unnecessary; you can just use the MITM-vulnerable `twisted.internet.ssl.ClientContextFactory` for now, or if #1368 gets merged first then use the real validation code.